src/ucs/debug/debug.c

```diff
@@ -302,42 +331,62 @@ static int get_line_info(struct backtrace_file *file, int backoff,
 
 /**
  * Create a backtrace from the calling location.
- */
-static void ucs_debug_backtrace_create(struct backtrace *bckt)
+ *
+ * @param bckt          Backtrace object.
+ * @param strip         How many frames to strip.
+*/
+void ucs_debug_backtrace_create(backtrace_h *bckt, int strip)
 {
+    size_t size = sizeof(**bckt);
     struct backtrace_file file;
     void *addresses[BACKTRACE_MAX];
     int i, num_addresses;
+    ucs_status_t status;
+
+    *bckt  = NULL;
+    status = ucs_mmap_alloc(&size, (void**)bckt, 0
+                            UCS_MEMTRACK_NAME("debug backtrace object"));
+    if (status != UCS_OK) {
+        ucs_assert(*bckt == NULL);
+        return;
+    }
 
     num_addresses = backtrace(addresses, BACKTRACE_MAX);
 
-    bckt->size     = 0;
-    bckt->position = 0;
+    (*bckt)->size       = 0;
+    (*bckt)->position   = strip;
+    (*bckt)->exclude    = 1;
+    (*bckt)->alloc_size = size;
```

Comment:
why need to save allocation size? isn't it constant?

Reply:
ucs_mmap_alloc returns real size allocated, usually it is rounded to page size, and this size should be used to free memory

Reply:
let's roundup in mmap_free as well then

Reply:
ok, done